cloudsync/tests/test_provider.py

```diff
@@ -295,25 +295,21 @@ def __filter_root(self, obj):
             if not self.prov.is_subpath(self.test_root, raw_path):
                 return False
 
-            self.__strip_root(obj)
-
         return True
 
     def __strip_root(self, obj):
-        if hasattr(obj, "path"):
-            path = obj.path
-            if path:
-                obj = copy.copy(obj)
-                relative = self.prov.is_subpath(self.test_root, path)
-                if not relative:
-                    # event from prior test
-                    return obj
-                path = relative
+        if hasattr(obj, "path") and obj.path:
+            relative = self.prov.is_subpath(self.test_root, obj.path)
+            if relative:
                 # TODO: This does not obey provider control over paths. Frex, consider windows paths and "C:"
-                if not path.startswith(self.prov.sep):
-                    path = self.prov.sep + path
-                obj.path = path
+                if not relative.startswith(self.prov.sep):
+                    relative = self.prov.sep + relative
+                # return a copy with modified path
+                obj_copy = copy.copy(obj)
+                obj_copy.path = relative
+                return obj_copy
```

Comment:
please ensure that you always return a copy, in order to avoid confusion about what is being returned